### PR TITLE
CentOS Support

### PR DESCRIPTION
**Pull request by athiele**
Here you go,

I squashed a few commits so it is easier for you to merge I hope.

Have a nice weekend!

### CODE REVIEW

File: templates/default/nginx_gitlab.erb [hunk 1]
```diff
@@ -8,14 +8,14 @@ upstream gitlab {
...
-  listen [::]:80;
+  listen 0.0.0.0:80;
   server_name <%= node['gitlab']['host'] %>;
   rewrite ^ https://$server_name$request_uri? permanent;
 }
 <%- end %>
 
 server {
-  listen [::]:<%= if node['gitlab']['ssl'] then '443' else '80' end %>;
+  listen 0.0.0.0:<%= if node['gitlab']['ssl'] then '443' else '80' end %>;
```
Comment (andruby) [inline templates/default/nginx_gitlab.erb:23]: Does the general `[::]` ipv4 and ipv6 matcher break on CentOS? I'd rather keep the codebase ipv4 and ipv6 compatible.

File: recipes/default.rb [hunk 2]
```diff
@@ -200,4 +227,19 @@
...
+nginx_site "gitlab"
+
+case node["platform"]
+when "centos"
+  bash "Open port 80 in firewall" do
+    code <<-EOH
+    iptables -I INPUT 1 -m state --state NEW -m tcp -p tcp --dport 80 -j ACCEPT
+    EOH
+  end
+end
```
Comment (andruby) [inline recipes/default.rb]: I don't think that this cookbook should handle firewall changes. I think it is better to handle it separately using the [firewall cookbook](https://github.com/opscode-cookbooks/firewall).

**Comment by andruby** [comment]
I'm tempted to merge this, but I'm not convinced yet that we should force rvm onto CentOS, Redhat & Fedora users..

I'll merge it when you remove the debug and firewall code. I'll add you to the README as maintainer for CentOS support :-)

File: recipes/default.rb [hunk 3]
```diff
@@ -200,4 +227,19 @@
   notifies :reload, "service[nginx]"
 end
 
-nginx_site "gitlab"
\ No newline at end of file
+bash "Fix permisions" do
+  code <<-EOH
+  chmod -R 755 /home/git
+  EOH
+end
```
Comment (andruby) [inline recipes/default.rb:228]: You can do this cleaner with chef's built in [directory resource](http://docs.opscode.com/resource_directory.html).

Comment (athiele) [inline recipes/default.rb:228]: Unfortunaley not: http://tickets.opscode.com/browse/CHEF-1621

File: templates/default/nginx_gitlab.erb (hunk 1 above)
Comment (athiele) [inline templates/default/nginx_gitlab.erb:23]: Unfortunately it does break CentOS. I will implement a switch here.

**Comment by athiele** [comment]
Hey Andrew,

thanks for the feedback. I will work it over soon.
Concerning the RVM discussion: There is no other way to get a 1.9 Ruby on CentOS with RPMs / yum. I am not the biggest RVM fan either, but I think this is the only clean way to get 1.9 ruby installed.

**Comment by andruby** [comment]
That sucks. Have you looked into using [ruby-build](https://github.com/sstephenson/ruby-build#installing-as-a-standalone-program-advanced) without rbenv? 

Frankly I would like to extract the whole ruby installation out of this cookbook so people can use whichever ruby they want. On Ubuntu 12.04 the situation is only slightly better as the official packages are stuck at 1.9.3p0. I want a newer ruby version too, but that should not be the responsibility of this cookbook.

**Comment by athiele** [comment]
ruby-build standalone does the same as rvm does - compile ruby from source. Thus I prefer rvm because it is more common.

**Comment by andruby** [comment]
For rvm to work however the rvm initiation shell code needs to run. And the gem lwrp doesn't work.

In an other cookbook I'm using fnichol's [chef-ruby_build](https://github.com/fnichol/chef-ruby_build).

``` ruby
include_recipe "ruby_build"
ruby_build_ruby "2.0.0-p195" do
  prefix_path "/usr/local/"
end
```

This builds ruby 2.0.0-p195 globally in /usr/local removing the need for rvm, rbenv or any other init code..

I'll add something similar in an optional recipe in this cookbook. This should also work for CentOS.

**Comment by athiele** [comment]
I just tried to use the ruby_build resource as you said, but the ruby installation is not there dureing the chef run. Could you give me a hint or a further example where you used it?

**Comment by andruby** [comment]
I'll merge it as is tonight. We can extract rvm specific code on a later date.

Thanks for trying ruby_build :-)

**Pull request merged**